### ISSUE TITLE
Kotlin native

### ISSUE THREAD

**Issue by mrmar096** [zxing-cpp/zxing-cpp#800]
I'm trying to get the kotlin-native dependency from maven central but is missing and I didn't see in github actions any release for kotlin-native.

**Comment by axxel** [comment]
You are right, that the binaries are not yet published on maven central. This is because I don't consider this to be 'released', yet. Having build artifacts uploaded after the CI build to the GHA pages, sounds like a good, though. But that would currently depend on @ISNing to provide that as I have no time for that a.t.m..

**Comment by mrmar096** [comment]
Yes, but I see you have the maven publish setup on the kotlin native wrapper, so what's left to finish? Can I help you? I'm really interested in the kotlin-native version to use in my KMP project.

**Comment by axxel** [comment]
Thanks for the offer. The package has not been published to maven because I don't consider it to be "released", yet. I wanted to let some time pass and see if there is any feedback which would suggest to change the API before 'setting it in stone' so to speak. In that regard, I'm happy to hear someone else is actively looking at the K/N wrapper and trying it out.

I intended to release it with the next release of the project. But I don't know when that is going to happen.

**Comment by natario1** [comment]
Would you consider releasing it under some `-alpha` version suffix? This makes it clear that the API surface may change, while also removing a barrier in adopting the library.

**Comment by axxel** [comment]
@ISNing to make some progress here and after I've done that for the android package, I decided to add the `-SNAPSHOT` suffix to the version string and run your publish-kn CI script for the first time. Unfortunatelly, it [failed](https://github.com/zxing-cpp/zxing-cpp/actions/runs/10763074538/job/29844188069), complainging that `Task 'publishReleasePublicationToSonatypeRepository' not found in root project 'zxing-cpp'.` Do you have any idea what might me missing here?

**Comment by axxel** [comment]
Thanks @ISNing. There are now a whole bunch of KotlinNative artifacts in version 2.2.1-SNAPSHOT available on maven: https://s01.oss.sonatype.org/#nexus-search;quick~zxing-cpp

You probably need to add a separate `repositories` entry in your `build.cradle` file:
```
maven { url = uri("https://s01.oss.sonatype.org/content/repositories/snapshots") }
```

If someone could confirm that those actually work, that would be much appreciated.

**Comment by axxel** [comment]
@mrmar096 @natario1 Does the published 2.2.1-SNAPSHOT package on mavenCentral fulfill your requirements?

**Comment by axxel** [comment]
Closing for lack of feedback from OP.
